### DISCUSSION TITLE
Can't resolve 'fs' at compile time in one js file, but not another, in the same folder.

### DISCUSSION THREAD

**Discussion by mbalcomm** [vercel/next.js#46574] [Help]
I have a rather strange situation: 
I am getting: 
```
wait  - compiling...
error - ./pages/api/utils.js:2:11
Module not found: Can't resolve 'fs'
  1 | const path = require('path')
> 2 | const fs = require('fs')
    |           ^
  3 | 

```
 
This file is in pages/api/ folder, whereas another js file with the same import works perfectly. 

I have tried set fs: false in next.config.js file and it did not solve my issue. 

I am using Nextjs 13. with App folder and this 'middleware' code is under pages/api folder. 
Thanks for any insightful information and help. 

M

**Comment by icyJoseph** [comment]
Who else imports from this file?

The problem might be that this file is being imported into a browser/edge bundle.

**Reply by mbalcomm** [reply]
There are only 3 *.js files in pages/api folders.  and on the app/ side, none 'use client' components import them.

**Reply by icyJoseph** [reply]
Mmm I'd need to see a repository or reproduction demo. 

You are not using Edge runtime or Turbo pack, etc?

**Comment by IgorZanellaDev** [comment]
I had the same problem, and I should know the problem. Maybe, you are importing other functions in that file in a client component. You have to divide into different files the functions you use for SSG and SSR and the ones you use in components.

Example:
utils.ts has function x (which use fs) and y (which doesn't use fs).
ComponentA.tsx uses function x in getStaticProps.
ComponentB.tsx uses function y in the client.

The error is that ComponentB loads utils.ts in the frontend, which has the fs import, and it doesn't work. As I said, you should divide the functions with fs in another file. 😉

**Reply by icyJoseph** [reply]
Exactly, some import is touching this file, within the wrong environment, but  it is often hard to see. Typically it happens when importing types, which you can fix by using the `import type` syntax... but OPs files are `.js`~ the only way to tell for sure is to take a look at the project, or project dependency graph
